state/charm.go

```diff
@@ -314,6 +331,35 @@ func safeConfig(ch charm.Charm) *charm.Config {
 	return escapedConfig
 }
 
+// safeLXDProfile ensures that the LXDProfile that we put into the mongo data
+// store, can in fact store the profile safely by escaping mongo-
+// significant characters in config options.
```

Comment:
BTW maps containing mongo-special characters are actually fine to put in documents unless you want to search on them. We've been doing it for a long time and there's no real issue with it AFAIK.